### PR TITLE
gh-99032: datetime docs: Encoding is no longer relevant

### PR DESCRIPTION
**Pull request by wjandrea**
The `.strftime()` methods return a Unicode string as of Python 3.0.

---

For reference, under the hood, the `.strftime()` methods use `time.strftime()`, whose return value is defined [here](https://github.com/python/cpython/blob/5893b5db98b38b17750c0572c7209774a5034898/Modules/timemodule.c#L881-L897) for the current tip of the main branch and [here](https://github.com/python/cpython/blob/3.0/Modules/timemodule.c#L535-L539) for 3.0.

---

Please check my work before merging to make sure I've understood what the docs are saying. I'm also not super familiar with C or RST, but I'm quite sure I've understood them correctly.

Backports should be made to all current versions.

---

BTW, encoding isn't mentioned in the `time` docs, so no changes required there I don't think.

* Issue: gh-99032

### CODE REVIEW

**Review by DanielNoord — APPROVED** [review]

**Review by AA-Turner — APPROVED** [review]

**Comment by merwok** [comment]
Could you create a ticket to describe the issue and make it easier to track the fixes?  Thanks

**Review by nanjekyejoannah — COMMENTED** [review]
I agree, maybe we need an issue to track this.

**Comment by wjandrea** [comment]
@merwok @nanjekyejoannah I could open an issue, sure, I'm just not sure what to say beyond what I've written here. Maybe,

> The docs say that `strptime()` will return an encoded string (i.e. `bytes`), but actually it returns a Unicode string as of Python 3.0 (i.e. `str`).

Actually, now that I think about it again, maybe it'd be better to change the warning from "may contain Unicode characters encoded ..." to "may contain non-ASCII characters." Say the word and I'll open an issue so we can discuss that.

**Comment by merwok** [comment]
That’s just the process we follow to keep things manageable 🙂 
A bug report doesn’t have to be very long, if the title is clear and the description tells clearly what the problem is, then we can apply labels to categorize the problem, note all versions that need a fix, people get notified about it, etc.
The issue describes the problem and the PR discussion can focus on the solution.  That’s all!

Full details here: https://devguide.python.org/triage/issue-tracker/

**Comment by Jason-Y-Z** [comment]
Thanks for the PR. I think you will need to include the issue number in your PR title :)

File: Doc/library/datetime.rst [hunk 1]
```diff
@@ -2496,11 +2496,7 @@ Notes:
 (1)
    Because the format depends on the current locale, care should be taken when
    making assumptions about the output value. Field orderings will vary (for
-   example, "month/day/year" versus "day/month/year"), and the output may
-   contain Unicode characters encoded using the locale's default encoding (for
-   example, if the current locale is ``ja_JP``, the default encoding could be
-   any one of ``eucJP``, ``SJIS``, or ``utf-8``; use :meth:`locale.getlocale`
-   to determine the current locale's encoding).
+   example, "month/day/year" versus "day/month/year").
```
Comment (pganssle) [inline Doc/library/datetime.rst]:
```suggestion
   example, "month/day/year" versus "day/month/year"), and the output may
   contain non-ASCII characters.
```

I think this change is a net positive, but I'm wondering if we might want to give people a head's up that they shouldn't assume that the output is ASCII, even though `str` is now a Unicode object? Maybe the world is at a point where we don't have to point this out, but maybe it won't hurt?

**Review by pganssle — CHANGES_REQUESTED** [review]

File: Doc/library/datetime.rst (hunk 1 above)
Comment (merwok) [reply Doc/library/datetime.rst]: I think this suggestion is a good idea!

**Review by pganssle — APPROVED** [review]

File: Doc/library/datetime.rst (hunk 1 above)
Comment (pganssle) [reply Doc/library/datetime.rst]: Good enough for me ☺

**Review by abalkin — APPROVED** [review]
LGTM

**Pull request merged**